pkgs/applications/science/programming/scyther/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, callPackage, callPackage_i686, buildEnv
+, includeGUI ? true
+, version ? "v1.1.3"
+}:
+let
+  source = (callPackage ./sources.nix {}).getSource "${version}";
+
+  cli = callPackage_i686 ./cli.nix { inherit source; };
+  gui = callPackage ./gui.nix { inherit source cli; };
+in
+  buildEnv {
+    name = "scyther-${version}";
+
+    paths = [ cli ] ++ lib.optional includeGUI gui;
+    pathsToLink = ["/bin"];
+
+    postBuild = if includeGUI then ''
```

Comment:
Ah yes, I'm copying the whole ./gui directory from the source to $out, because that's where all the GUI is. I believe there is some way for the GUI to show these protocols, that's why they're there. But the CLI wouldn't include them, but because the protocols are so useful, I decided to have them in there anyways. Could be made a seperate output though, doesn't technically need to be there.